core/src/main/java/org/elasticsearch/action/search/SearchPhaseExecutionException.java

```diff
@@ -85,7 +85,13 @@ private static Throwable deduplicateCause(Throwable cause, ShardSearchFailure[]
     @Override
     public RestStatus status() {
         if (shardFailures.length == 0) {
-            // if no successful shards, it means no active shards, so just return SERVICE_UNAVAILABLE
+            // If we have a cause that can report a REST status code then return that.
+            // One such failure occurs when a globally unmapped field is searched with strict search mode on.
+            Throwable cause = unwrapCause();
+            if (cause instanceof ElasticsearchException) {
+                return ((ElasticsearchException) getCause()).status();
+            }
+            // otherwise when no unsuccessful shards, it means no active shards, so just return SERVICE_UNAVAILABLE
```

Comment:
I was getting 500 errors despite deliberately throwing a 400-status ParsingException from the FetchSearchPhase.innerRun() method.
The general exception handling logic in place looks to assume that all exceptions thrown without a single shard failure must be an internal 500 error which is clearly not the case here. I toyed with the idea of making my strict test fabricating a single-shard failure and passing that through somehow but thought it might be better to change the central error handling to use whatever status flag might be set on the failing exception.